File: xiang.li/1222.kt
import java.util.*

class Solution {
    fun queensAttacktheKing(queens: Array<IntArray>, king: IntArray): List<List<Int>> {
        val chessboard = Array(8) { CharArray(8) }
        for (queen in queens) {
            chessboard[queen[0]][queen[1]] = 'q'
        }
        val directions = arrayOf(intArrayOf(0, 1), intArrayOf(0, -1), intArrayOf(1, 0), intArrayOf(-1, 0), intArrayOf(1, 1), intArrayOf(-1, -1), intArrayOf(1, -1), intArrayOf(-1, 1))
        val res: MutableList<List<Int>> = LinkedList()
        for (direction in directions) {
            dfs(chessboard, king, res, direction)
        }
        return res
    }

    private fun dfs(chessboard: Array<CharArray>, cur: IntArray, container: MutableList<List<Int>>, direction: IntArray) {
        if (!verify(cur[0], cur[1])) {
            return
        }
        if (chessboard[cur[0]][cur[1]] == 'q') {
            val temp: MutableList<Int> = LinkedList()
            temp.add(cur[0])
            temp.add(cur[1])
            container.add(temp)
            return
        }
        dfs(chessboard, intArrayOf(cur[0] + direction[0], cur[1] + direction[1]), container, direction)
    }

    private fun verify(i: Int, j: Int): Boolean {
        return i in 0..7 && j >= 0 && j < 8
    }
}

File: hang.liu/1222.js
var queensAttacktheKing = function(queens, king) {
    var length = 8
    var result = [];
    var options = [-1, 0, 1];
  
    var optionsLength = options.length;
    for (var j = 0; j < optionsLength; j++) {
      for (var k = 0; k < optionsLength; k++) {
        for (var i = 1; i <= length; i++) {
          var temp = [king[0] + i * options[j], king[1] + i * options[k]];
          if (temp[0] < 0 || temp[0] >= length || temp[1] < 0 || temp[1] >= length) continue 
          if (queens.filter(item => item[0] === temp[0] && item[1] === temp[1]).length) {
            result.push(temp);
            break;
          }
        }
      }
    }
      
   return result
  };